src/machine/machine_atsamd51.go

```diff
@@ -1733,36 +1734,94 @@ func (pwm PWM) getMux() PinMode {
 
 // USBCDC is the USB CDC aka serial over USB interface on the SAMD21.
 type USBCDC struct {
-	Buffer *RingBuffer
+	Buffer  *RingBuffer
+	TxIdx   volatile.Register8
+	waitTxc bool
+	sent    bool
 }
 
-// WriteByte writes a byte of data to the USB CDC interface.
-func (usbcdc USBCDC) WriteByte(c byte) error {
-	// Supposedly to handle problem with Windows USB serial ports?
+const (
+	usbcdcTxSizeMask uint8 = 0x3F
+	usbcdcTxBankMask uint8 = ^usbcdcTxSizeMask
+	usbcdcTxBank1st  uint8 = 0x00
+	usbcdcTxBank2nd  uint8 = usbcdcTxSizeMask + 1
+)
+
+// Flush flushes buffered data.
+func (usbcdc *USBCDC) Flush() error {
 	if usbLineInfo.lineState > 0 {
-		// set the data
-		udd_ep_in_cache_buffer[usb_CDC_ENDPOINT_IN][0] = c
+		idx := usbcdc.TxIdx.Get()
+		sz := idx & usbcdcTxSizeMask
+		bk := idx & usbcdcTxBankMask
+		if 0 < sz {
+
+			if usbcdc.waitTxc {
+				// waiting for the next flush(), because the transmission is not complete
+				return nil
+			}
+			usbcdc.waitTxc = true
 
-		usbEndpointDescriptors[usb_CDC_ENDPOINT_IN].DeviceDescBank[1].ADDR.Set(uint32(uintptr(unsafe.Pointer(&udd_ep_in_cache_buffer[usb_CDC_ENDPOINT_IN]))))
+			// set the data
+			usbEndpointDescriptors[usb_CDC_ENDPOINT_IN].DeviceDescBank[1].ADDR.Set(uint32(uintptr(unsafe.Pointer(&udd_ep_in_cache_buffer[usb_CDC_ENDPOINT_IN][bk]))))
+			if bk == usbcdcTxBank1st {
```

Comment:
If you change the lines above as I was suggesting, this line can be:

```go
if idx & usbcdcTxBankMask == usbcdcTxBank1st {
```

Reply:
L1765 also uses bk, so it would be easier to understand if you don't change it.
Would it be better to change it?